spec/master_manipulator/log_spec.rb

```diff
@@ -0,0 +1,128 @@
+require 'spec_helper'
+require 'fileutils'
+
+describe MasterManipulator::Log do
+
+  let(:beaker_host)           { instance_double(Beaker::Host) }
+  let(:dummy_class)           { Class.new { extend MasterManipulator::Log} }
+#  let(:beaker_command)        { instance_double(Beaker::Command) }
+  let(:log_dir)               { "/tmp/MasterManipulator_Log/#{Process.pid}/puppetlabs/puppet" }
+  let(:log_dir_to_make)       { "/tmp/MasterManipulator_Log/#{Process.pid}/puppetlabs/puppetserver" }
+  let(:log_file)              { "/tmp/MasterManipulator_Log/#{Process.pid}/puppetlabs/puppetserver/puppetserver.log" }
+  let(:backup_log_file_base)  { "/tmp/MasterManipulator_Log/#{Process.pid}/puppetlabs/puppetserver/puppetserver.1976-07-04" }
+  let(:cmd_1)                 { 'puppet config print logdir' }
+  let(:cmd_test_log_file)     { "test -f #{log_file}" }
+  let(:cmd_date)              { "date +%Y-%m-%d" }
+  let(:cmd_test_backup_log_file_generic) { "test -f #{backup_log_file_base}.*" }
+  let(:cmd_test_backup_log_file_0) { "test -f #{backup_log_file_base}.0.log" }
+  let(:cmd_test_backup_log_file_1) { "test -f #{backup_log_file_base}.1.log*" }
+  let(:max_trys)              { 100 }
+
+
+  def shared_dsl_expectations
+    log_result = Beaker::Result.new('host', 'cmd')
+    log_result.stdout = log_dir
+    expect(dummy_class).to receive(:on).with(beaker_host, cmd_1).and_return(log_result)
+
+    date_result = Beaker::Result.new('host', 'cmd')
+    date_result.stdout = "1976-07-04"
+    expect(dummy_class).to receive(:on).with(beaker_host, cmd_date).and_return(date_result)
+  end
+  
+#add test for if log file is missing
+
+  context 'normal' do 
```

Comment:
Or I could call it "All test for the log module" to be more descriptive.  I am kinda fuzzy on what it ends up being used for besides the one case of version compatibility.  So it is hard to pick something that might be relevant in the future if a second context comes into being.